jmclient/jmclient/blockchaininterface.py

```diff
@@ -430,59 +499,14 @@ def query_utxo_set(self, txout, includeconfs=False, include_mempool=True):
                 result.append(result_dict)
         return result
 
-    def get_unspent_indices(self, transaction):
-        """ Given a CTransaction object, identify the list of
-        indices of outputs which are unspent (returned as list of ints).
-        """
-        bintxid = transaction.GetTxid()[::-1]
-        res = self.query_utxo_set([(bintxid, i) for i in range(
-            len(transaction.vout))])
-        # QUS returns 'None' for spent outputs, so filter them out
-        # and return the indices of the others:
-        return [i for i, val in enumerate(res) if val]
-
-    def estimate_fee_per_kb(self, N):
-        """ The argument N may be either a number of blocks target,
-        for estimation of feerate by Core, or a number of satoshis
-        per kilo-vbyte (see `fee_per_kb_has_been_manually_set` for
-        how this is distinguished).
-        In both cases it is prevented from falling below the current
-        minimum feerate for tx to be accepted into node's mempool.
-        In case of failure to connect, None is returned.
-        In case of failure to source a specific minimum fee relay rate
-        (which is used to sanity check user's chosen fee rate), 1000
-        is used.
-        In case of failure to source a feerate estimate for targeted
-        number of blocks, a default of 10000 is returned.
-        """
-
-        # use the local bitcoin core minimum mempool fee as floor to avoid
-        # relay problems
-        rpc_result = self._rpc('getmempoolinfo', None)
+    def _get_mempool_min_fee(self) -> Optional[int]:
```

Comment:
Is this new?

Reply:
Yes, moved advanced JM fee logic `estimate_fee_per_kb()` to base class, as it should be the same for all blockchain interfaces (like feerate randomizations, etc), so needed new methods instead of calling `._rpc()` directly.